### PR TITLE
Match wheel tags against `Requires-Python` major-minor

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Given `Requires-Python: ">=3.12.3"`, we were rejecting wheels like `dearpygui-1.11.1-cp312-cp312-win_amd64.whl`, since `3.12.0` is not included in `>=3.12.3`. We instead need to test against the major-minor version of `Requires-Python`.

The easiest way to do this, I think, is the use the `RequiresPython` struct, which has a single bound that we can truncate the major-minor. This also means that we now allow `dearpygui-1.11.1-cp312-cp312-win_amd64.whl` for specifiers like `Requires-Python: "==3.10.*"`. This is incorrect on the surface, but it does match our semantics for `Requires-Python` elsewhere: we treat it as a lower bound.

Closes https://github.com/astral-sh/uv/issues/5287.

### CODE REVIEW

File: crates/uv-resolver/src/requires_python.rs [hunk 1]
```diff
@@ -160,6 +160,23 @@ impl RequiresPython {
...
+    pub fn bound_major_minor(&self) -> RequiresPythonBound {
+        match self.bound.as_ref() {
+            // Ex) `>=3.10.1` -> `>=3.10`
+            Bound::Included(version) => RequiresPythonBound(Bound::Included(Version::new(
+                version.release().iter().take(2),
+            ))),
+            // Ex) `>3.10.1` -> `>=3.10`
+            // This is unintuitive, but `>3.10.1` does indicate that _some_ version of Python 3.10
+            // is supported.
+            Bound::Excluded(version) => RequiresPythonBound(Bound::Included(Version::new(
```
Comment (charliermarsh) [inline crates/uv-resolver/src/requires_python.rs:173]: This is strange but I think correct...

Comment (konstin) [reply crates/uv-resolver/src/requires_python.rs:174]: Yeah `>3.10` is a footgun because it does actually include `3.10` version, this is basically PYI006

**Comment by konstin** [comment]
> This also means that we now allow dearpygui-1.11.1-cp312-cp312-win_amd64.whl for specifiers like Requires-Python: "==3.10.*"

For warehouse with `==3.11.*`, this increases the lockfile from 4161 lines to 4466 lines (+7%).

**Review by konstin — APPROVED** [review]

**Comment by charliermarsh** [comment]
> For warehouse with ==3.11.*, this increases the lockfile from 4161 lines to 4466 lines (+7%).

Yeah, unfortunately I think we kinda have to do this, unless we want to change our `Requires-Python` semantics. But we should do that holistically if so.

**Comment by konstin** [comment]
The most correct solution is probably computing an upper bound the same way we compute the lower bound, incl. widening to the next full major-minor version/discarding the patch version, and then only applying the lower bound for matching dependencies' requires-python, but applying the lower and the upper bound for selecting their wheels. (Giving us requires-python semantic #3, i think).

Improving this isn't schema-breaking or badly breaking backwards compatibility (the wheels can never be selected anyway), so we can move this post-release-ready.

**Pull request merged**